Sources/_SwiftSyntaxTestSupport/SyntaxProtocol+Initializer.swift

```diff
@@ -58,40 +58,40 @@ private class InitializerExprFormat: BasicFormat {
 }
 
 private extension TriviaPiece {
-  var initializerExpr: ExprSyntaxProtocol {
+  var initializerExpr: ExprSyntax {
     let (label, value) = Mirror(reflecting: self).children.first!
     switch value {
     case let value as String:
-      return FunctionCallExpr(callee: ".\(label!)") {
+      return ExprSyntax(FunctionCallExpr(callee: ".\(label!)") {
```

Comment:
I'm tempted to add a `.asBaseType` (a default method on `Decl/Stmt/Expr/PatternSyntaxProtocol`) or something, thoughts?

Reply:
If we want this as a postfix notation (and I think it might be nice). I think it should be `as(ExprSynax.self)`. But don’t most of these conversions now become unnecessary that we have the generic initializers?

Reply:
It would have to be `cast(ExprSyntax.self)` since `as` returns optional and it would be nice if we didn't have to specify `ExprSyntax` at all since we know this node's base type.

The conversion is necessary *because* of the generic initializers, unless the callsite uses `fromProtocol` instead 🤷. Unless I'm missing something, which is entirely possible.